Create recipes for TED by topics

benoit74 opened the pull request:
## Rationale

- Automatically create one TED recipe per TED topic

This is a maintenance script. It is not expected to be used on a regular basis but still useful to keep / share.

rgaudin commented on dispatcher/backend/maint-scripts/create_ted_topics_recipes.py:
```diff
@@ -0,0 +1,152 @@
+            "task_name": "ted",
+            "warehouse_path": "/.hidden/dev",
+        },
+        "enabled": True,
```
Is this what we want? Run them all so Content Team can review them?

benoit74 replied:
This is what we wanted at least

rgaudin reviewed (changes requested):
Can you explain what this is for, what will be the Content-Team actions following running this script?

- Will those be individually checked? What's the strategy regarding title/description metadata? Do you want to run and fail all those that will be above 30/80c?
- What about the `mul` and static list of languages? Do all topics have all those languages? In that order of importance? If not, how will this be acknowledged and fixed by content team?
- With enabled=True we are talking about hundreds of runs that will need to be checked. Alternative is to enabled=False and let Content team run/review at their own pace.
- How many recipes is this?

benoit74 commented:
> Can you explain what this is for, what will be the Content-Team actions following running this script?

This is the script which **has been** used to generate all TED recipes by topics. The goal is only to not loose it so that it might be reused later on if needed.

See https://github.com/openzim/zim-requests/issues/789 as well for answers to some of your questions.

>  Will those be individually checked? What's the strategy regarding title/description metadata? 

They will be individually checked before being move to "prod" (library.kiwix.org)

> Do you want to run and fail all those that will be above 30/80c?

I don't understand this question, sorry

> What about the mul and static list of languages? Do all topics have all those languages? In that order of importance? If not, how will this be acknowledged and fixed by content team?

Good point, it reminded me I forgot to create some issues in TED scraper.

This is kinda of hack to retrieve videos in all languages (it is not possible to say "all" in languages, see https://github.com/openzim/ted/issues/171).

Clearly not all topics have those languages, and order is not handled either (the scraper should filter and order them, see https://github.com/openzim/ted/issues/172)

I did think this is going to be acknowledged and fixed by content team, I consider this should be fixed by dev team ; doing it it manually while it is mostly straightforward to automate with code is a bit sad

> With enabled=True we are talking about hundreds of runs that will need to be checked. Alternative is to enabled=False and let Content team run/review at their own pace.

This has already ran and allowed to proceed quickly ^^

>  How many recipes is this?

355 recipes, 355 ZIMs

benoit74 commented on dispatcher/backend/maint-scripts/create_ted_topics_recipes.py:
```diff
@@ -0,0 +1,152 @@
+
+def create_recipe(ted_topic_name: str, access_token):
+    clean_ted_topic_name = ted_topic_name.replace(" ", "-")
+    schedule_name = f"ted_topic_{clean_ted_topic_name}"
```
There is no convention on schedule name AFAIK. Content team decided it is better without language code in this case 🤷🏻‍♂️

rgaudin replied:
🤯

benoit74 commented on dispatcher/backend/maint-scripts/create_ted_topics_recipes.py:
```diff
@@ -0,0 +1,152 @@
+    ted_topics = get_ted_topics()
+    logger.debug(ted_topics)
+    for topic in ted_topics:
+        if topic != "street art":
```
This is the only one which is going to be executed in fact.

I decided to keep this in the script as a foolproof thing to ensure the next person who will use the script will understand how it works before launching it, and if not it will impact only one topic/recipe

rgaudin commented:
> This is the script which has been used to generate all TED recipes by topics. The goal is only to not loose it so that it might be reused later on if needed.

Well if it's to be kept for reference, don't request a review!

> Do you want to run and fail all those that will be above 30/80c?

I was wondering if you wanted to check the length of all those title/desc with the script or just let it run and have the scraper fail if it didn't fit. I understand from the last answer that all recipes created their ZIM so all metadata did fit.

> the scraper should filter and order them, see https://github.com/openzim/ted/issues/172

👍 I think this should be fixed before moving to prod if we are sending this to `Languages` metadata (readers trust this)
